src/classes/UTIL_Namespace.cls

```diff
@@ -97,23 +104,48 @@ public class UTIL_Namespace {
     * @param str token name 
     * @return token name, with namespace prefix removed, if present.
     ********************************************************************************************************/
-    public static string StrTokenRemoveNSPrefix(string str) {
-        if(str.startsWith('npsp__'))
-            str = str.replaceFirst('npsp__', '');
-        if (getNamespace() != '' && str.startsWith(getNamespace() + '__'))
-            str = str.replaceFirst(getNamespace() + '__', '');
+    public static String StrTokenRemoveNSPrefix(String str) {
+        if (str != null) {
+            if (str.startsWith(NPSP_NS + '__')) {
+                str = str.replaceFirst(NPSP_NS + '__', '');
+            }
+                
+            if(getNamespace() != '' && str.startsWith(getNamespace() + '__')) {
+                str = str.replaceFirst(getNamespace() + '__', '');
+            }
+        }
+            
+        return str;
+    }
+
+    /*******************************************************************************************************
+    * @description Removes NPSP class prefix or dynamically removes current NS prefix.
+    * @param str : the class name to have the namespace removed.
+    * @return str : class name with namespace prefix removed.
+    ********************************************************************************************************/
+    public static String StrTokenRemoveClassNSPrefix(String str) {
```

Comment:
I think the namespace DOES actually belong in the class name here - as with the field names, we are presenting the  information in the context that is most relevant to the users, which means we are including the namespace on our class in our managed config - it's how an admin would see this class. While rare, there could also be a naming collision with a un-namespaced class provided by a customer org, for instance "BDI_CustObjMappingGAUAllocation", and our managed version "npsp.BDI_CustObjMappingGAUAllocation"

Reply:
@force2b  Yes StrTokenRemoveClassNSPrefix is needed because the removeClassNamespace method just strips any namespace without regard to what it was.  StrTokenRemoveClassNSPrefix only removes the NPSP namespace or the namespace currently applied to the UTIL_Namespace class.